airflow/api/common/experimental/get_tasks.py

```diff
@@ -34,11 +33,5 @@ def get_dag_run_state(dag_id, execution_date):
     # Get DAG object and check Task Exists
     dag = dagbag.get_dag(dag_id)
 
-    # Get DagRun object and check that it exists
-    dagrun = dag.get_dagrun(execution_date=execution_date)
-    if not dagrun:
-        error_message = ('Dag Run for date {} not found in dag {}'
-                         .format(execution_date, dag_id))
-        raise DagRunNotFound(error_message)
-
-    return {'state': dagrun.get_state()}
+    # Return the task.
+    return dag.task_ids
```

Comment:
Why do we need a function for `get_dag(dag_id).task_ids`? That seems... verbose.

Reply:
What do you mean? We need to return the task_ids and make sure that the dag_id passed does exist. I don't understand this comment fully.

Reply:
I mean that we remove this function, and where it is currently called in your PR:
```python
        task_list = list()
        task_ids = get_tasks(dag_id)
```

change it to something like.

```python
        task_ids = get_dag(dag_id).task_ids
```

Reply:
What you are saying is correct, but then we would have to get DagBag in the endpoints.py file and also do the verification of dag_id existance. I was also consiously trying not to touch models from this layer, and defer to all DB access tasks to the helper function defined in the api/common/experimental folder. This implementation seems cleaner to me, but if you guys would say otherwise I will change it.

Reply:
The change as I proposed won't need any changes anywhere else. Try it? I don't why you'd need to touch the DagBag at all - `get_dag()` will return a DAG object or will thrown an exception